driver-core/src/main/java/com/datastax/driver/core/Cluster.java

```diff
@@ -1504,18 +1504,7 @@ long delayMs() {
             Set<Host> contactPointHosts = Sets.newHashSet(allHosts);
 
             try {
-                try {
-                    controlConnection.connect();
-                } catch (UnsupportedProtocolVersionException e) {
-                    logger.debug("Cannot connect with protocol {}, trying {}", e.getUnsupportedVersion(), e.getServerVersion());
-
-                    connectionFactory.protocolVersion = e.getServerVersion();
-                    try {
-                        controlConnection.connect();
-                    } catch (UnsupportedProtocolVersionException e1) {
-                        throw new DriverInternalError("Cannot connect to node with its own version, this makes no sense", e);
-                    }
-                }
+                connect();
```

Comment:
I like that a lot of logic was moved into separate methods, makes code a lot easier to follow 👍